src/value.rs

```diff
@@ -0,0 +1,215 @@
+    };
+}
+
+#[derive(Copy, Clone, Debug)]
+pub struct EasingHandle {
+    pub x: f64,
+    pub y: f64,
+}
+
+/// Time for a particular keyframe, represented as a frame number.
+#[derive(Copy, Clone, Debug)]
+pub struct Time {
+    /// Frame number.
+    pub frame: f64,
+    /// Easing tangent going into the next keyframe
+    pub in_tangent: Option<EasingHandle>,
+    /// Easing tangent leaving the current keyframe
+    pub out_tangent: Option<EasingHandle>,
+    /// Whether it's a hold frame.
+    pub hold: bool,
+}
+
+impl Time {
+    /// Returns the frame indices and interpolation weight for the given frame,
+    /// and whether to hold the frame
+    pub(crate) fn frames_and_weight(
+        times: &[Time],
+        frame: f64,
+    ) -> Option<([usize; 2], f64, Easing, bool)> {
+        if times.is_empty() {
+            return None;
+        }
+        use core::cmp::Ordering::*;
+        let ix = match times.binary_search_by(|x| {
+            if x.frame < frame {
+                Less
+            } else if x.frame > frame {
+                Greater
+            } else {
+                Equal
+            }
+        }) {
+            Ok(ix) => ix,
+            Err(ix) => ix.saturating_sub(1),
+        };
+        let ix0 = ix.min(times.len() - 1);
+        let ix1 = (ix0 + 1).min(times.len() - 1);
+
+        let t0 = times[ix0];
+        let t1 = times[ix1];
+        let (t0_ox, t0_oy) = t0.out_tangent.map(|o| (o.x, o.y)).unwrap_or((0.0, 0.0));
+        let (t1_ix, t1_iy) = t1.in_tangent.map(|i| (i.x, i.y)).unwrap_or((1.0, 1.0));
+        let easing = Easing {
+            o: EasingHandle { x: t0_ox, y: t0_oy },
+            i: EasingHandle { x: t1_ix, y: t1_iy },
+        };
+        let hold = t0.hold;
+        let t = (frame - t0.frame) / (t1.frame - t0.frame);
+        Some(([ix0, ix1], t.clamp(0.0, 1.0), easing, hold))
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct Animated<T: Tween> {
+    pub times: Vec<Time>,
+    pub values: Vec<T>,
+}
+
+impl<T: Tween> Animated<T> {
+    /// Returns the value at the specified frame.
+    pub fn evaluate(&self, frame: f64) -> T {
+        self.evaluate_inner(frame).unwrap_or_default()
+    }
+
+    fn evaluate_inner(&self, frame: f64) -> Option<T> {
+        let ([ix0, ix1], t, easing, hold) = Time::frames_and_weight(&self.times, frame)?;
+        let t = if hold { 0.0 } else { t };
+
+        let v1 = self.values.get(ix0)?;
+        let v2 = self.values.get(ix1)?;
+
+        Some(v1.tween(v2, t, &easing))
+    }
+}
+
+/// Something that can be interpolated with an easing function.
+pub trait Tween: Clone + Default {
+    fn tween(&self, other: &Self, t: f64, easing: &Easing) -> Self;
+}
+
+impl Tween for f64 {
+    fn tween(&self, other: &Self, t: f64, _easing: &Easing) -> Self {
+        // TODO: We are enforcing linear interpolation for now, but a decent amount of work is done for easings.
+        keyframe::ease(keyframe::functions::Linear, *self, *other, t)
```

Comment:
That's correct. Everything was a lerp at one point, but then I added easings and captured those, but we aren't using them yet. I planned on finishing this up, but I think I ragequit trying to implement position keyframe easing, since it's non-trivial and underdocumented in the lottie spec.

Essentially position keyframes are supposed to be `Point(a.x.ease(b.x) + pos.x, a.y.ease(b.y) + pos.y)` but there was something going on elsewhere that made it hard to narrow down